docs/reference/commandline/build.md

```diff
@@ -373,6 +374,21 @@ the command line.
 > repeatable builds on remote Docker hosts. This is also the reason why
 > `ADD ../file` will not work.
 
+### Build with Secrets
+
+Build Secrets can be specified to provide private data such as keys or
+passwords.
+
+Build an image using secrets:
+
+```bash
+docker build -t app \
+    --secret source=/path/to/github-key,target=git.key .
+```
+
+This will expose the contents of `github-key` in the container during build
+as `/run/secrets/git.key`.
+
```

Comment:
@thaJeztah it was probably using the cache as by default the mode is read-only, no-execute.  Here is an example:

```
docker build --no-cache -t testimage --build-secret src=/usr/local/bin/docker,target=docker,mode=0755 .
Sending build context to Docker daemon 12.49 MB
Step 1/4 : FROM alpine:latest
 ---> 88e169ea8f46
Step 2/4 : RUN mount | grep /run
 ---> Running in d158b3ff377c
tmpfs on /run type tmpfs (ro,relatime)
 ---> 9a3f370992b0
Removing intermediate container d158b3ff377c
Step 3/4 : RUN ls -lah /run/secrets/
 ---> Running in f99f664b5034
total 12200
drwxr-xr-x    2 root     root          60 Feb  7 19:52 .
drwxrwxrwt    3 root     root          60 Feb  7 19:52 ..
-rwxr-xr-x    1 root     root       11.9M Feb  7 19:52 docker
 ---> d12b13566935
Removing intermediate container f99f664b5034
Step 4/4 : RUN /run/secrets/docker -v
 ---> Running in 2c64f083244f
Docker version 1.14.0-dev, build 46c4cda71
 ---> 8e94d203c623
Removing intermediate container 2c64f083244f
Successfully built 8e94d203c623
```
In the original issue we decided that an explicit cache bust was preferable instead of busting or trying to maintain a HASH of a secret (https://github.com/docker/docker/pull/28079#issuecomment-260211452)